tests/utils/test_environment.py

```diff
@@ -244,6 +244,23 @@ def test_process_pip_requirements(tmpdir):
     assert reqs == ["mlflow==1.2.3"]
     assert cons == []
 
+    # Ensure a requirement for mlflow is preserved when package hashes are specified
+    conda_env, reqs, cons = _process_pip_requirements(
+        ["a"],
+        pip_requirements=[
+            """mlflow==1.20.2 \
+    --hash=sha256:963c22532e82a93450674ab97d62f9e528ed0906b580fadb7c003e696197557c \
+    --hash=sha256:b15ff0c7e5e64f864a0b40c99b9a582227315eca2065d9f831db9aeb8f24637b"""
+        ],
+    )
```

Comment:
```suggestion
    hash1 = "963c22532e82a93450674ab97d62f9e528ed0906b580fadb7c003e696197557c"
    hash2 = "b15ff0c7e5e64f864a0b40c99b9a582227315eca2065d9f831db9aeb8f24637b"
    conda_env, reqs, cons = _process_pip_requirements(
        ["a"], pip_requirements=[f"mlflow==1.20.2 --hash=sha256:{hash1} --hash=sha256:{hash2}"],
    )
```

Can we store hashes in variables to improve the readability?

Reply:
Test refactored